cms/serializers.py

```diff
@@ -164,11 +165,41 @@ def get_page_url(self, instance):
     def get_price(self, instance):
         return instance.price[0].value["text"] if len(instance.price) > 0 else None
 
+    def get_financial_assistance_form_url(self, instance):
```

Comment:
This is less bad, I think - still not sure if there's a better way to grab the children forms - 
I COULD grab a `values_list` off the first query to determine the `selected_program` conditionals - there's also apparently a `specific()` method from wagtail that will return the most subclassed page object rather than the generic page, through fewer queries, but I'm not sure whether or not that will help here.I can easily do the first bit (grab a list of tuples and work with that rather than call `selected_program_id` twice), but I'm not well versed in the ways of wagtail and worry I'll make the `get_children()` call worse.

Reply:
Something like 
```
        fin_form_related_program_ids = [program.id for program in instance.program.related_programs]
        fin_form_program_ids = fin_form_related_program_ids + [instance.program.id]
        flex_pricing_forms = (FlexiblePricingRequestForm.objects.filter(live=True).values_list("id", "selected_program_id", "slug", ))
        fin_assistance_pages_by_program_ids = (FlexiblePricingRequestForm.objects.filter(
            selected_program_id__in=fin_form_program_ids
        )).live().values_list("id", "selected_program_id", "slug", "get_parent_product_page__id").order_by("-id")
        selected_fin_assistance_page = next(
            (x for n, x in enumerate(fin_assistance_pages_by_program_ids) if x[1] == instance.program.id), None
        )
        if selected_fin_assistance_page is None:
            selected_fin_assistance_page = instance.get_children().type(FlexiblePricingRequestForm).live().first()

        if selected_fin_assistance_page is None and (len(instance.program.related_programs) > 0):
            selected_fin_assistance_page = next(
                (x for n, x in enumerate(fin_assistance_pages_by_program_ids) if x[1] in fin_form_related_program_ids), None
            )
        return(
            f"{instance.get_url()}{selected_fin_assistance_page[2]}/"
            if selected_fin_assistance_page is not None
            else ""
        )


```

Reply:
I'd say just leave it for now cause we want to get this out today. I was more making a note about it.